gph/python/ripser_interface.py

```diff
@@ -447,7 +459,16 @@ def ripser_parallel(X, maxdim=1, thresh=np.inf, coeff=2, metric="euclidean",
     if metric == 'precomputed':
         dm = X
     else:
-        dm = pairwise_distances(X, metric=metric, **metric_params)
+        if not issparse(X) and thresh != np.inf and \
+                metric in ['euclidean', 'minkowski']:
```

Comment:
@ulupo this is a draft, I do not think that this solution is easy to scale up if we need to support more metrics.

The other issue is that `cKDTree` does not support sparse input. I choose to not go with the optimized solution if the input is sparse.

Let me know if you have a different solution/approach.

Reply:
This looks reasonable for now.